server/lib/chat/chat-models.ts

```diff
@@ -1,77 +1,21 @@
 import sql, { SQLStatement } from 'sql-template-strings'
-import { ServerChatMessageType } from '../../../common/chat'
-import { SbUserId } from '../../../common/users/user-info'
+import { ChannelPermissions, ServerChatMessageType } from '../../../common/chat'
+import { SbUser, SbUserId } from '../../../common/users/user-info'
 import db, { DbClient } from '../db'
 import transact from '../db/transaction'
 import { Dbify } from '../db/types'
 
-export interface UserChannelsEntry {
-  channelName: string
-  joinDate: Date
-}
-
-type DbUserChannelsEntry = Dbify<UserChannelsEntry>
-
-export async function getChannelsForUser(userId: SbUserId): Promise<UserChannelsEntry[]> {
-  const { client, done } = await db()
-  try {
-    const result = await client.query<DbUserChannelsEntry>(sql`
-      SELECT channel_name, join_date
-      FROM joined_channels
-      WHERE user_id = ${userId}
-      ORDER BY join_date`)
-    return result.rows.map(row => ({ channelName: row.channel_name, joinDate: row.join_date }))
-  } finally {
-    done()
-  }
-}
-
-export interface ChannelUsersEntry {
-  userId: SbUserId
-  userName: string
-  joinDate: Date
-}
-
-type DbChannelUsersEntry = Dbify<ChannelUsersEntry>
-
-export async function getUsersForChannel(channelName: string): Promise<ChannelUsersEntry[]> {
-  const { client, done } = await db()
-  try {
-    const result = await client.query<DbChannelUsersEntry>(sql`
-      SELECT u.id AS user_id, u.name AS user_name, c.join_date
-      FROM joined_channels as c INNER JOIN users as u ON c.user_id = u.id
-      WHERE c.channel_name = ${channelName}
-      ORDER BY c.join_date`)
-    return result.rows.map(row => ({
-      userId: row.user_id,
-      userName: row.user_name,
-      joinDate: row.join_date,
-    }))
-  } finally {
-    done()
-  }
-}
-
-export interface ChannelPermissions {
-  kick: boolean
-  ban: boolean
-  changeTopic: boolean
-  togglePrivate: boolean
-  editPermissions: boolean
-  owner: boolean
-}
-
-export interface JoinedChannel {
+export interface UserChannelEntry {
   userId: SbUserId
   userName: string
```

Comment:
Currently it's only used when adding a new user to the channel. Maybe we can adjust the return type of that function to something like `UserChannelEntry & { userName: string }`, but not sure if that's better.

Reply:
I can see from looking at the code that it's just constructing an `SbUser` from the stuff returned here, you should just stop doing that and request the user out of the DB separately like we do elsewhere. It will be a lot more resilient to us adding things to the user structure that need to be sent down as well